bin/lib/check_reqs.js

```diff
@@ -362,6 +362,36 @@ module.exports.run = function () {
     });
 };
 
+/** Checks if Windows SDK required to build the target_platform is present
+ * @param {String}  target_platorm        Target platform ('8.1' or '10.0')
+ * @return {Boolean}                      Whether the min SDK required to build the platform is present  
+ *
+ */ 
+module.exports.isWinSDKPresent = function (target_platform) {
+    return getInstalledWindowsSdks()
+    .then(function (installedSdks) {
+        var reqVersion = Version.tryParse(REQUIRED_VERSIONS[target_platform].windowssdk);
+        var hasSdkInstalled = installedSdks.some(function (installedSdk) {
```

Comment:
It seems like ```getInstalledWindowsSdks()``` doesn't return an array in all cases, i.e: when an exception occurs while running the registry query. Maybe, modify it to return an empty array in such a case, otherwise we'll error out here.